### PR TITLE
[location] Make foreground service permission opt-in

### PR DESCRIPTION
**Pull request by brentvatne**
# Why

See: #26846

tl;dr: Google Play now requires declaring usage of foreground service permissions, and you must include a video to demonstrate the usage.

# How

- Removed the permission from **AndroidManifest.xml** so it is not added by default. I chose this approach over adding the permission to the block list in the config plugin, because this may impact apps that use the permission elsewhere! This would be very confusing if it were to impact someone. The tradeoff is that if someone is using expo-location in a bare app (without CNG) and did not add the permissions as stated in the README, but rather depended on the foreground service permission being implicitly added via the library **AndroidManifest.xml**, then updating to a new expo-location version would lead to this permission being missing.
- Updated Android permissions data using @byCedric's scripts, which I published as an npm package and added as a script here to make it a bit easier in the future (`yarn permissions-sync-android` in docs).
- [x] Backport docs changes to SDK 50 if we decide to cherrypick this change

# Test Plan

@alanjhughes is going to verify that leaving out these permissions will have no unintended side effects.

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [x] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Review by alanjhughes — APPROVED** [review]
This should be good. Most of the location functionality still works as normal. The only difference is requesting background location permission will always fail.
